airbyte-metrics/reporter/src/main/java/io/airbyte/metrics/reporter/ToEmit.java

```diff
@@ -59,4 +61,22 @@
     this(toEmit, 15, TimeUnit.SECONDS);
   }
 
+  /**
+   * Wrapper callable to handle 1) query exception logging and 2) counting metric emissions so
+   * reporter app can be monitored too.
+   *
+   * @param metricQuery
+   * @return
+   */
+  private static Runnable countMetricEmission(Procedure metricQuery) {
+    return () -> {
+      try {
+        metricQuery.call();
+        DogStatsDMetricSingleton.gauge(MetricsRegistry.EST_NUM_METRICS_EMITTED_BY_REPORTER, 1);
```

Comment:
Should this be a counter instead of a gauge? We are not counting anything here since we always report 1. Maybe the method should be named reportMetricEmission instead.

Reply:
+1 Gauge would make sense if we are recording a value in time.  However, this looks like it is simply counting the execution.

Reply:
Thanks for catching that. This is what happens when I program at 2 am, whoops.